src/main/java/util/LottoNumberGenerator.java

```diff
@@ -1,33 +1,35 @@
 package util;
 
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.List;
+import domain.LottoNumber;
+
+import java.util.*;
 
 public class LottoNumberGenerator {
 
-    private static List<Integer> allNumbers;
     public static final Integer MIN_LOTTO_NUMBER = Integer.valueOf(1);
     public static final Integer MAX_LOTTO_NUMBER = Integer.valueOf(45);
     public static final Integer LOTTO_NUMBER_COUNT_LIMIT = Integer.valueOf(6);
 
+    private static final Map<Integer, LottoNumber> lottoNumberCache = new HashMap<>();
+
     static {
         initialize();
     }
 
-    private LottoNumberGenerator() {
-    }
-
     private static void initialize() {
-        allNumbers = new ArrayList<>();
         for (int i = MIN_LOTTO_NUMBER; i <= MAX_LOTTO_NUMBER; i++) {
-            allNumbers.add(i);
+            lottoNumberCache.put(i, new LottoNumber(i));
         }
     }
 
     public static List<Integer> generateLottoNumber() {
```

Comment:
반환타입을 Integer가 아니라 LottoNumber로 해도 되지 않을까요? 
```suggestion
    public static List<LottoNumber> generateLottoNumber() {
```